cosmos/game-of-stakes
# Remove the cartel

This PR removes a cartel from the GoS which currently holds 53% of the voting power in GoS.

## Rationale

Blog post: https://medium.com/certus-one/uncovering-a-game-of-stakes-cartel-f895d9591da1
Additional proof by @joepindar: https://medium.com/@joepindar_22415/identify-potential-cartel-relay-network-in-cosmos-gos3-6c995d69ce06 (amazing work :tada:)

- They hold 53% of the bonded stake, steadily increasing with each validator that gets jailed.
- Centralizing such an amount of voting power in a densely connected network within a single provider results in an unrealistic network.
- This was evidenced by the spam transaction-induced network halt, which was caused by them all crashing at once.
- It also sped up the decentralized start - the network launched when they brought their nodes online.
- They did a sybil attack on the registration process, which is against the rules.

We ask the Cosmos community to corroborate our findings, validate our evidence and genesis.json, as well as post any extra evidence they might have.

This PR introduces the `game_of_stakes_5` chain, with a genesis date set to 15:00 UTC on Saturday in order to give everyone time to discuss and verify this proposal. We ask anyone who supports this proposal not to bring up their validators on the `game_of_stakes_4` chain. 

Please state your support by giving a "thumbs up" on this post, as well as adding a comment with the moniker you represent.

## How the kicking process works

The liberation software does the following:
* It instantly unbonds all delegations from the cartel validators
* It instantly unbonds all outgoing delegations from cartel members
* It transfers the balances of all cartel members to the CommunityPool

## How to reproduce

1. Clone the `liberation` branch [here](https://github.com/certusone/cosmos-sdk/tree/liberation)
2. Build the cosmos-sdk as usual
3. Run the command **1** on a host that has the game_of_stakes_3 chain data using the binary that you just built
4. Copy the output into `export.json` in the `genesisify` of the `game_of_stakes` repo
5. Follow the instructions in the `genesisify` folder's README (from this PR as we changed the vesting addrs.json to remove the cartel)
6. Change the `chain_id` to `game_of_stakes_5` and the `genesis_time` to `2019-02-02T15:00:00Z`
7. You should now have the *same* genesis.json (except for sorting - use `jq . --sort-keys` to allow a better diff)

Command **1**:
```
./gaiad export --height 330000 --for-zero-height --kick cosmosvaloper102p00zgyzs2xr9tvt3ju9jnhdtwuft83jngvtu,cosmosvaloper1z770dde04y6zlanhsdvgscsawjzkynmmpqm8we,cosmosvaloper1pxuk5c3q9dr79300xfv0yafjf9stf3w788zrms,cosmosvaloper1snppu44r8p3apmfkwu0ujxcjkmdzw59zjq2pgq,cosmosvaloper1z3yluu93wa9yfv2sfrvk5m2q9c3s343qk4q48w,cosmosvaloper1drtdtat7x6pqhfs9u90c0czvg65lwtqhawyx2j,cosmosvaloper1lnm4q9yfpmxkkndcjzjntzu9kqxew400vhtgjd,cosmosvaloper15xxqq8v4q0sqcfhuqtflncddfep5jxxpqudtuk,cosmosvaloper13zyvm5hrh7yqasg04ffd9e4etpeakttkvdpea2,cosmosvaloper1uk7wd0psudl83cgzw6khzjnnen6tkyzc9hmhzs,cosmosvaloper1d9v2wvuzcuqwl3euhzy8p3khvlveqsf970072r,cosmosvaloper1evh32gcrtgmw43g9a9s9u9yxmswpx8cfcpvpq3,cosmosvaloper1wmc09y9uxhyylhv0trv5xyr6u82lrttl4alxza,cosmosvaloper1frs78q7gcv8s2cd7pnnqk09slyf69pdxmj22dw,cosmosvaloper1njvv9cd0avfzrkn04xyh3weu4dsuhv5mdj530c,cosmosvaloper17s2ns29hkcsgqp7rl5n9rckx9hnss84zuzvyv9,cosmosvaloper1epta0djtp5gdakthj4pwugvww85l6tpfnzu8p6,cosmosvaloper18zrap0tnzt9c0vnkqflefzmcgg5cg434uh5ryd,cosmosvaloper1sjx8jxuplf6ajcvxwtpntpjm5s64xt3u8er35c,cosmosvaloper1l98y3tyl3h6ez5xevh8dgaymr5spdrwqnlqwde,cosmosvaloper1tgnqq2g9u0syzlnuh9v96evnlkllv6f8jednf0,cosmosvaloper1f2t99m5zenwasugr3tmw58djsmqlrj2087u9tg,cosmosvaloper17ewzayt2edwtghnrjdl58ufhqcgmw48rlyuz9v,cosmosvaloper14dp76w9srztynh40m2zm2hlaykpxtnm235enmz,cosmosvaloper17060h323px5swd3mj2p0cf9zxpecg9vnk4nrls,cosmosvaloper1jkdr05fuud9ne6dsm25wxfge7x9e57f7u8a2ej,cosmosvaloper15qavwkx8qv8y9htxdppamsjfu4f5627jv5r9k7,cosmosvaloper13wpyugwm3cxt58cmvy08up54pa47dpj0u9tdx8,cosmosvaloper1tva47y9h9zh4llgau6qsz4c2wzf0zkpxnvzf38,cosmosvaloper1unvmn2wsh65avenajzfy254j0v8d58f7x85a7r,cosmosvaloper1msesrv5jxt0jk5vtwrgygdjq60vnn6a4lazaqn,cosmosvaloper1m9285yt4l994xread77druww85r9g8j6q535nm,cosmosvaloper1t3da02katujzgqnpnulew6v9xqmzj756lp6nzt,cosmosvaloper1caruqk82j6chyswjkp28u3sk80ya5zla2uhwjj,cosmosvaloper1ujqf7m03qd6gf7car0teefgc50qujtzc3dn0dr,cosmosvaloper1hw7nu0ssa7huk0zv3za0xgnmcxm7ryd78l7gdp,cosmosvaloper1f2579m7lk7k94nm2qna4462sq63nrqkl46cysj,cosmosvaloper1r4punjg5t3zsx0nmfk5lackpy622c8fj5q5j50,cosmosvaloper1uuzqhrcjfaeusa5cx0ljjjkszqmxheslau2tyw,cosmosvaloper16vylggc7cwwjyucedk2250ugcgjlnka8d0zpyc,cosmosvaloper1xf6upkfzgl8f3pkpa0l2d76pex33ur27mjqezp,cosmosvaloper123ahlc3vkfe8hf354gdcddr3lzxdmyg2x09cfk,cosmosvaloper1la3s8xgn5h3nzjdgsdhgdkf29n9ud0kmmad2nt,cosmosvaloper1u332h2zq5z3hc8yva967svs0tdf98h0qshd709,cosmosvaloper14t77a9pzc6fel0nsrlsfp3ft937ltrf6w0xdfe,cosmosvaloper1vf6mdfr6y3wstmve96c63xps9sc70sw7e78kpj,cosmosvaloper1f797cgxtj4ueyvzlras08tq6zn6l87ckdctlkm,cosmosvaloper16fu02hmnhlvtf47gv92mz0mkejjvhcqw2axgxf,cosmosvaloper198ssnhd4p547r7vwfrse64v307gynvj52cmu4e,cosmosvaloper1sc4f5y2xhwgn4hjjqg983q83tcgdf3m5vxx86v,cosmosvaloper13knf79lcspvv0vmtkjvssncud4vqxzzv97dy3j,cosmosvaloper1je7k9c5s2wvzrxnm4dzhamx9rayugqas3t3hkx,cosmosvaloper1dxa36kdp8a7nlls8k7k2m5rt7zzv8qhfywn6r7,cosmosvaloper1l7pmprwv6yyxr9pk7ttyuwfyclyxf9gznrcgr2,cosmosvaloper19t5k4la55fr3s92e4hj5uqpk956ka6uzlzrzc7,cosmosvaloper1fcsy7vpzvaaqlu5cvezfzjt3s4z663xdmapn56,cosmosvaloper1jq8ml9x52sxww4mwfzk58t5dw4a566hxf5ctqr,cosmosvaloper1gux94xvmj6uwf0ncpxhzqnlmrm9rayleqwa07h,cosmosvaloper1v88fcs0cn7tyhgmy94evkaarv9gkqwrle4cxu9,cosmosvaloper13crr30dyx70g6jqvvwr8caeha65aare9uzzqja,cosmosvaloper1hnlarnp8gtdn9rff6xzpl99z64d9755sy0gdan,cosmosvaloper174km4mr509mtcm8kt7thlxvrtksk4u88xedj84,cosmosvaloper1gpwyat78vagv5hujuqyteajkmgq3wku57lf0f0,cosmosvaloper12u2e2qn74zlk34g223l7jthx53t2wgcjzuqn74,cosmosvaloper1haxtrc2enmesgcp6ltj824s623jx8xlp2fvcku,cosmosvaloper1lzm8cv5c4ezgd0vsc5ker5y5veumxh4t8xpnc5,cosmosvaloper1f797cgxtj4ueyvzlras08tq6zn6l87ckdctlkm,cosmosvaloper197034kwpjxm0y3umeuxk3tvnw7vedf05l58ue6,cosmosvaloper1lp7y8kck2swwhurt2tm2svj28e9dwk45q7wj2v,cosmosvaloper12t65dsk4fdnxylj2wtr0x8qcgjthvvegqg7zkm,cosmosvaloper1apd0w3eywwr3jsl9f2mlqm7z8ymrhxsvmahynu,cosmosvaloper16tflhzzrjy4w4s9reky35ft7tqu8cglka82zac,cosmosvaloper1c27f7klwxze75phrd0nnqwddpva3gzq8j9zt6p
```
This genesis file has been tested against [this](https://github.com/cosmos/cosmos-sdk/tree/cwgoes/gos-genesis-port-working) branch using `make test_sim_gaia_custom_genesis_fast`
